src/uct/api/uct.h

```diff
@@ -693,11 +693,17 @@ ucs_status_t uct_iface_get_address(uct_iface_h iface, uct_iface_addr_t *addr);
  * it does not detect issues such as network mis-configuration or lack of connectivity.
  *
  * @param [in]  iface      Interface to check reachability from.
- * @param [in]  addr       Address to check reachability to.
+ * @param [in]  dev_addr   Device address to check reachability to.
+ * @param [in]  ep_addr    Remote endpoint address to check reachability to.
+ * @param [in]  iface_addr Interface address to check reachability to.
+ *
+ * @note One or more addresses can be NULL depends on interface specific.
+ *       If all addresses are NULL the return value will be 0.
```

Comment:
Thank you. I will talk to Yossi about this.
In case of the question "What if I pass NULL and the remote EP is not null" I would say that the caller algorithm don't want to connect to `EP` but want to connect to `iface`. As I understand, this is the main idea in having these two addresses (`dev_add` and `iface_addr`) - to provide better flexibility in connection procedure. In one hand, we have algorithm which wants to select interface/ep to work and, in other hand, we have the interface which should decide workability with the algorithm. If the interface gets NULL in the pointer it required - the answer `false` (I don't want to be selected).
Sorry, perhaps I'm wrong and I need to learn an other interfaces logic more.